recipes/gtsam/all/conanfile.py

```diff
@@ -64,16 +64,17 @@ class GtsamConan(ConanFile):
         "enable_consistency_checks": False,
         "install_cppunitlite": True,
         "install_matlab_toolbox": False,
-        "pose3_expmap": False,
-        "rot3_expmap": False,
+        "pose3_expmap": True,
+        "rot3_expmap": True,
         "slow_but_correct_betweenfactor": False,
-        "support_nested_dissection": False,
-        "tangent_preintegration": False,
+        "support_nested_dissection": True,
+        "tangent_preintegration": True,
         "throw_cheirality_exception": True,
         "use_quaternions": False,
-        "with_TBB": False,
+        "with_TBB": True,
         "with_eigen_MKL": False,
         "with_eigen_MKL_OPENMP": False,
+        "with_vendored_metis": True,  # Cannot be disabled in C3I due to a non-default option value
```

Comment:
This comment is not clear to me. I think the common practice in CCI is to try to use dependencies from conan as much as possible. In this case, if possible, we should try to avoid adding an option for this and make the metis requirement conditional to `support_nested_dissection`. That way we would simplify the recipe and avoid leaving the decision of using vendored metis to the user (better user experience in the end).

Let me know what you think or point out anything I might be missing. Thanks!

Reply:
It references this overridden option value for METIS:
```python
        # GTSAM expects 32-bit integers for Metis
        self.options["metis"].with_64bit_types = False
```
I kind of agree, but the METIS and nested dissection support is pretty important for this library, unfortunately. Disabling it effectively removes parallelization support. So I would leave it as is for now and instead look into improving either GTSAM or METIS to get rid of this deficiency.

Reply:
Looking at the METIS recipe again, it looks like it should really be defaulting to 32-bit types in the first place to match the default behavior of the library's build system: https://github.com/KarypisLab/METIS/commit/81d70ba3d799c0143e38085b4cdd4e891e39b49a
I'll open a PR.

Reply:
@valgur thanks a lot for the explanation and quick search. Indeed, we could change the default value in the metis recipe and make the requirement of metis compulsory (unvendored) in this gtsam recipe 👏

Reply:
https://github.com/conan-io/conan-center-index/pull/23194